### ISSUE TITLE
After installed the .net core 2.1 rc, "dotnet ef" doesn't work for the ".net core 1.1" project

### ISSUE THREAD

**Issue by TonyWoo** [dotnet/core#1532]
# Issue Title

After installed the .net core 2.1 rc, dotnet ef doesn't work for the ".net core 1.1" project

# General
I just installed the .net core 2.1 rc (dotnet-sdk-2.1.300-rc1-008673-win-x64.exe), now I cannot run the "dotnet ef" for my .net core 1.1 project, it shows below error:

"No executable found matching command "dotnet-ef"

Can anyone help on this issue? Thanks.

**Comment by leecow** [comment]
@ajcvickers @livarcocc - PTAL

**Comment by livarcocc** [comment]
I believe this is a known breaking on dotnet ef and the recommendation is to pin yourself using global.json to 2.1.200 (latest 2.0 released version of the SDK).

@ajcvickers can confirm add details if necessary, I believe.

**Comment by ajcvickers** [comment]
Yes, that's correct. Although I am surprised at the exception since I thought there was work to generate a specific error message here? /cc @bricelam

**Comment by livarcocc** [comment]
You get the error message if you have a DotnetCliToolReference in your project. Given that this is failing to execute dotnet ef, I imagine the user is missing that reference.

@TonyWoo can you share the output of dotnet --info when executed within your project folder?

**Comment by TonyWoo** [comment]
@livarcocc please see below output of dotnet --info
>dotnet --info
.NET Command Line Tools (1.1.5)

Product Information:
 Version:            1.1.5
 Commit SHA-1 hash:  2b517ddd29

Runtime Environment:
 OS Name:     Windows
 OS Version:  10.0.16299
 OS Platform: Windows
 RID:         win10-x64
 Base Path:   C:\Program Files\dotnet\sdk\1.1.5

Host (useful for support):
  Version: 2.1.0-rc1
  Commit:  eb9bc92051

.NET Core SDKs installed:
  1.1.5 [C:\Program Files\dotnet\sdk]
  2.0.0 [C:\Program Files\dotnet\sdk]
  2.0.3 [C:\Program Files\dotnet\sdk]
  2.1.2 [C:\Program Files\dotnet\sdk]
  2.1.104 [C:\Program Files\dotnet\sdk]
  2.1.105 [C:\Program Files\dotnet\sdk]
  2.1.200 [C:\Program Files\dotnet\sdk]
  2.1.300-rc1-008673 [C:\Program Files\dotnet\sdk]

.NET Core runtimes installed:
  Microsoft.AspNetCore.All 2.1.0-rc1-final [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.All]
  Microsoft.AspNetCore.App 2.1.0-rc1-final [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 1.0.8 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 1.1.5 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 2.0.0 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 2.0.3 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 2.0.5 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 2.0.6 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 2.0.7 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 2.1.0-rc1 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]

To install additional .NET Core runtimes or SDKs:
  https://aka.ms/dotnet-download

------------------------------------
I set the sdk version to 1.1.5 in global.json
{
  "sdk": {
    "version": "1.1.5"
  }
}

If I set it to 2.1.200, it will throw below error when run dotnet ef

NuGet.Packaging.Core.PackagingException: Unable to find fallback package folder 'C:\Program Files (x86)\Microsoft SDKs\NuGetPackagesFallback'.
   at NuGet.Packaging.FallbackPackagePathResolver..ctor(String userPackageFolder, IEnumerable`1 fallbackPackageFolders)
   at Microsoft.DotNet.Cli.Utils.LockFileExtensions.GetPackageDirectory(LockFile lockFile, LockFileTargetLibrary library)
   at Microsoft.DotNet.Cli.Utils.PackagedCommandSpecFactory.GetCommandFilePath(LockFile lockFile, LockFileTargetLibrary toolLibrary, LockFileItem runtimeAssembly)
   at Microsoft.DotNet.Cli.Utils.PackagedCommandSpecFactory.CreateCommandSpecFromLibrary(LockFileTargetLibrary toolLibrary, String commandName, IEnumerable`1 commandArguments, IEnumerable`1 allowedExtensions, LockFile lockFile, CommandResolutionStrategy commandResolutionStrategy, String depsFilePath, String runtimeConfigPath)
   at Microsoft.DotNet.Cli.Utils.ProjectToolsCommandResolver.ResolveCommandSpecFromToolLibrary(SingleProjectInfo toolLibraryRange, String commandName, IEnumerable`1 args, IProject project)
   at Microsoft.DotNet.Cli.Utils.ProjectToolsCommandResolver.ResolveCommandSpecFromAllToolLibraries(IEnumerable`1 toolsLibraries, String commandName, IEnumerable`1 args, IProject project)
   at Microsoft.DotNet.Cli.Utils.ProjectToolsCommandResolver.ResolveFromProjectTools(CommandResolverArguments commandResolverArguments)
   at Microsoft.DotNet.Cli.Utils.ProjectToolsCommandResolver.Resolve(CommandResolverArguments commandResolverArguments)
   at Microsoft.DotNet.Cli.Utils.CompositeCommandResolver.Resolve(CommandResolverArguments commandResolverArguments)
   at Microsoft.DotNet.Cli.Utils.CommandResolver.TryResolveCommandSpec(ICommandResolverPolicy commandResolverPolicy, String commandName, IEnumerable`1 args, NuGetFramework framework, String configuration, String outputPath, String applicationName)
   at Microsoft.DotNet.Cli.Utils.Command.Create(ICommandResolverPolicy commandResolverPolicy, String commandName, IEnumerable`1 args, NuGetFramework framework, String configuration, String outputPath, String applicationName)
   at Microsoft.DotNet.Cli.Utils.Command.Create(String commandName, IEnumerable`1 args, NuGetFramework framework, String configuration, String outputPath, String applicationName)
   at Microsoft.DotNet.Cli.Program.ProcessArgs(String[] args, ITelemetry telemetryClient)
   at Microsoft.DotNet.Cli.Program.Main(String[] args)

Please note I have below reference in my project file:
  <ItemGroup>
    <DotNetCliToolReference Include="Microsoft.EntityFrameworkCore.Tools.DotNet" Version="1.0.1" />
  </ItemGroup>

Thanks for everyone's help.

**Comment by TonyWoo** [comment]
Hi @livarcocc , thanks for your help. I think I found the cause. 

1) Installed the .net core 2.1 rc, 
2) I tried to run the dotnet ef,  because I didn't place the global.json file in my project root folder, I think it tried to use the 2.1 to restore the project and then failed. but it generated some project.assets.json in the obj folder. 
3) I have to create the global.json and set it to 1.1.5
4) Now when I run dotnet ef, it looks it cannot restore due to the previous project.assets.json is there. but it just show the error ""No executable found matching command "dotnet-ef"

Since the global.json is there and have the SDK pinned to 1.1.5, to solve this issue, I just delete the whole obj folder and then I can run dotnet ef successfully.

Thanks all for the help.
